### PR TITLE
[Internal] Samples: Adds OpenTelemetry and Application Insights samples

### PR DESCRIPTION
**Pull request by jcocchi**
## Description

Adds a sample project for how to configure OpenTelemetry instrumentation and a sample project showing how to configure Application Insights instrumentation. 

## Type of change

- [x] This change requires a documentation update

### CODE REVIEW

File: Microsoft.Azure.Cosmos.Samples/Usage/ApplicationInsights/Program.cs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+                string aiConnectionString = configuration["ApplicationInsightsConnectionString"];
+                if (string.IsNullOrEmpty(authKey) || string.Equals(authKey, "Super secret connection string"))
+                {
+                    throw new ArgumentException("Please specify a valid ApplicationInsightsConnectionString in the appSettings.json");
+                }
+
+                // <SetUpApplicationInsights>
+                IServiceCollection services = new ServiceCollection();
+                services.AddLogging(loggingBuilder => loggingBuilder.AddFilter<ApplicationInsightsLoggerProvider>("Azure-Cosmos-Operation-Request-Diagnostics", LogLevel.Information));
```
Comment (sourabh1007) [inline Microsoft.Azure.Cosmos.Samples/Usage/ApplicationInsights/Program.cs:49]: You don't need this line...it should work without this line also.

File: Microsoft.Azure.Cosmos.Samples/Usage/OpenTelemetry/Program.cs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+                // <SetUpOpenTelemetery>
+                ResourceBuilder resource = ResourceBuilder.CreateDefault().AddService(
+                            serviceName: serviceName,
+                            serviceVersion: "1.0.0");
+
+                // Set up logging to forward logs to chosen exporter
+                using ILoggerFactory loggerFactory = LoggerFactory.Create(builder => builder.AddOpenTelemetry(options =>
+                    {
+                        options.IncludeScopes = true;
```
Comment (sourabh1007) [inline Microsoft.Azure.Cosmos.Samples/Usage/OpenTelemetry/Program.cs:57]: do we need this scopes flag to be set? if it is working without it, let's remove this config.

File: Microsoft.Azure.Cosmos.Samples/Usage/OpenTelemetry/OpenTelemetry.csproj [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    <AssemblyName>Cosmos.Samples.OpenTelemetry</AssemblyName>
+    <RootNamespace>Cosmos.Samples.OpenTelemetry</RootNamespace>
+    <LangVersion>latest</LangVersion>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Azure.Monitor.OpenTelemetry.Exporter" Version="1.0.0-beta.10" />
+    <PackageReference Include="Microsoft.Azure.Cosmos" Version="3.32.3-preview" />
+    <PackageReference Include="Microsoft.Extensions.Azure" Version="1.6.3" />
+    <PackageReference Include="OpenTelemetry" Version="1.5.0-alpha.2" />
```
Comment (sourabh1007) [inline Microsoft.Azure.Cosmos.Samples/Usage/OpenTelemetry/OpenTelemetry.csproj:15]: Use opentelemetry latest stable if Azure exporter is fine with it.

File: Microsoft.Azure.Cosmos.Samples/Usage/OpenTelemetry/OpenTelemetry.csproj [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net7.0</TargetFramework>
+    <AssemblyName>Cosmos.Samples.OpenTelemetry</AssemblyName>
+    <RootNamespace>Cosmos.Samples.OpenTelemetry</RootNamespace>
+    <LangVersion>latest</LangVersion>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Azure.Monitor.OpenTelemetry.Exporter" Version="1.0.0-beta.10" />
+    <PackageReference Include="Microsoft.Azure.Cosmos" Version="3.32.3-preview" />
```
Comment (sourabh1007) [inline Microsoft.Azure.Cosmos.Samples/Usage/OpenTelemetry/OpenTelemetry.csproj:13]: both projects are referring to different versions of cosmosdb sdk. It should be same version of sdk (whichever is older)

File: Microsoft.Azure.Cosmos.Samples/Usage/OpenTelemetry/OpenTelemetry.csproj [hunk 5]
```diff
@@ -0,0 +1,28 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net7.0</TargetFramework>
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos.Samples/Usage/OpenTelemetry/OpenTelemetry.csproj:5]: Same comment here, unless net7 is the minimum required, use net6

File: Microsoft.Azure.Cosmos.Samples/Usage/ApplicationInsights/Program.cs (hunk 1 above)
Comment (jcocchi) [reply Microsoft.Azure.Cosmos.Samples/Usage/ApplicationInsights/Program.cs]: removed

File: Microsoft.Azure.Cosmos.Samples/Usage/OpenTelemetry/Program.cs (hunk 2 above)
Comment (jcocchi) [reply Microsoft.Azure.Cosmos.Samples/Usage/OpenTelemetry/Program.cs]: removed

File: Microsoft.Azure.Cosmos.Samples/Usage/OpenTelemetry/OpenTelemetry.csproj (hunk 3 above)
Comment (jcocchi) [reply Microsoft.Azure.Cosmos.Samples/Usage/OpenTelemetry/OpenTelemetry.csproj]: switched to the latest stable version

File: Microsoft.Azure.Cosmos.Samples/Usage/OpenTelemetry/OpenTelemetry.csproj (hunk 4 above)
Comment (jcocchi) [reply Microsoft.Azure.Cosmos.Samples/Usage/OpenTelemetry/OpenTelemetry.csproj]: downgraded both to the minimum required version

File: Microsoft.Azure.Cosmos.Samples/Usage/ApplicationInsights/ApplicationInsights.csproj [hunk 6]
```diff
@@ -0,0 +1,25 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net7.0</TargetFramework>
```
[the remark this answers is not in the document]
Comment (jcocchi) [reply Microsoft.Azure.Cosmos.Samples/Usage/ApplicationInsights/ApplicationInsights.csproj]: switched to net6

File: Microsoft.Azure.Cosmos.Samples/Usage/OpenTelemetry/OpenTelemetry.csproj (hunk 5 above)
Comment (jcocchi) [reply Microsoft.Azure.Cosmos.Samples/Usage/OpenTelemetry/OpenTelemetry.csproj]: switched to net6

File: Microsoft.Azure.Cosmos.Samples/Usage/OpenTelemetry/OpenTelemetry.csproj [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    <PackageReference Include="Microsoft.Extensions.Azure" Version="1.6.3" />
+    <PackageReference Include="OpenTelemetry" Version="1.5.0-alpha.2" />
+    <PackageReference Include="System.Diagnostics.DiagnosticSource" Version="7.0.2" />
+    <PackageReference Include="Microsoft.Extensions.Configuration" Version="3.1.0" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.FileExtensions" Version="2.2.0" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="2.2.0" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <None Include="AppSettings.json" Link="AppSettings.json">
```
[the remark this answers is not in the document]
Comment (jcocchi) [reply Microsoft.Azure.Cosmos.Samples/Usage/OpenTelemetry/OpenTelemetry.csproj:23]: switched to be consistent

**Review by ealsur — APPROVED** [review]

**Review by sourabh1007 — APPROVED** [review]

**Comment by sourabh1007** [comment]
> ## Description
> Adds a sample project for how to configure OpenTelemetry instrumentation and a sample project showing how to configure Application Insights instrumentation.
> 
> ## Type of change
> * [x]  This change requires a documentation update

It is not mandatory comment, but it will be good if we can have some screenshots generated from these samples on this PR. It will give us a feeling what to expect from these samples actually.

**Pull request merged**